package/gluon-status-page/files/lib/gluon/status-page/view/status-page.html

```diff
@@ -64,6 +64,20 @@
 		return v and translate('enabled') or translate('disabled')
 	end
 
+	local function formatBits(v, factor)
+		--to format a value given in kilobits use a factor of 1000
+		factor = factor or 1
+		units = {'k', 'M', 'G'}
+		units[0] = ''
```

Comment:
I don't like what you are doing here... By making 0 an index it is no longer a sequence as tables in Lua start with index 1. It is a bad example for other people even though it works for your code.

Reply:
I learned the other day this would be the proper one-line solution for my intention:
```lua
units = {[0]='','k', 'M', 'G'}
```

I haven't tested it on gluons lua version, yet.

However this is perfectly fine lua, which makes the following calculation lighter by one offset, right?

Reply:
@AiyionPrime It's not fine Lua although it works. It's not a sequence anymore then. In your case there is no good reason for using this "hack" and you are giving a bad example.

Reply:
You could just write:
`string.format('%s %sbit', bits, units[known_pow+1])`

Reply:
You would make people believe they can just do this trick in any case which will lead to erroneous behavior. Therefore it is not fine, but **rude** Lua...

Reply:
By the way. I tried to make a joke if that was not obvious...

Reply:
Which part?

Reply:
> I don't like what you are doing here... By making 0 an index it is no longer a sequence as tables in Lua start with index 1. It is a bad example for other people even though it works for your code.

Not that I aimed for it to be one, in fact this is unrelated, but I think it actually is a sequence.
Whatever the joke is supposed to be I don't get it. Is this resolved?

_Originally documented in section 3.4.7 https://www.lua.org/manual/5.3/manual.html_

Reply:
No, the joke was "**rude** Lua".

Excerpt from https://www.lua.org/pil/3.6.html
```
For those that really want their arrays starting at 0, it is not difficult to write the following:

    days = {[0]="Sunday", "Monday", "Tuesday", "Wednesday",
            "Thursday", "Friday", "Saturday"}

Now, the first value, "Sunday", is at index 0. That zero does not affect the other fields, but "Monday" naturally goes to index 1, because it is the first list value in the constructor; the other values follow it. Despite this facility, I do not recommend the use of arrays starting at 0 in Lua. Remember that most functions assume that arrays start at index 1, and therefore will not handle such arrays correctly. 
```

There is no strong reason for using for doing this **hack** in your code as you could just write `string.format('%s %sbit', bits, units[known_pow+1])` which is even shorter. Therefore you give a bad example.